silence/silence.go

```diff
@@ -0,0 +1,57 @@
+package silence
+
+import (
+	"fmt"
+	"github.com/go-chat-bot/bot"
+	"log"
+	"strconv"
+	"time"
+)
+
+var (
+	silentMap map[string]time.Time
+)
+
+func silenceFilter(cmd *bot.FilterCmd) (string, error) {
+	until, found := silentMap[cmd.Target]
+	if !found || time.Now().After(until) {
+		return cmd.Message, nil
+	}
+	log.Printf("Silencing message in %s\n", cmd.Target)
+	return "", nil
+}
```

Comment:
I think you could simplify this using [time.After](https://golang.org/pkg/time/#After), you can set a single boolean "silenced = true", and unsilence if after the time has passed.

The way it is now, the silentMap will only grow. To leave it this way I think you will need a goroutine to clean this map from time to time.

Reply:
Well the map will only have one item per channel at most. The silencing needs to be done per-channel so I think it's sensible. I *could* wipe the entry from silenceMap after time elapses though. I'd say it's not necessary most likely